examples/hello_services/android/app/src/main/java/com/example/flutter/Plugin.java

```
@@ -51,16 +52,17 @@ abstract protected boolean initialize(String channelName, JSONObject arguments,
         abstract protected boolean dispose(String channelName, FlutterView flutterView);
 
         void installInto(final FlutterView flutterView) {
-            flutterView.addOnMessageListener("listen" + name,
-                new FlutterView.OnMessageListener() {
+            flutterView.addOnMessageListenerAsync("listen" + name,
+                new FlutterView.OnMessageListenerAsync() {
                     @Override
-                    public String onMessage(FlutterView view, String json) {
+                    public void onMessage(FlutterView view, String json,
+                        FlutterView.MessageResponse response) {
                         String channelName = name + counter;
                         JSONObject arguments;
                         try {
                             arguments = new JSONObject(json);
                         } catch (JSONException e) {
-                            return null;
+                            return;
```

What happens to response in this case? Looks like it's just sitting there, waiting forever. Should we maybe return an error?